elastic/kibana
# [AO] Add evaluation values for metric threshold and inventory rules

Closes #153877

## Summary

This PR adds a new field called `kibana.alert.evaluation.values` to the alert document for metric threshold and inventory rules. This is an array of numbers but depending on the result of the rule execution, the value might be `null` too.

![image](https://user-images.githubusercontent.com/12370520/230380396-fcfa10d8-a119-497b-bd94-9f567ecb8fc5.png)

We want to use this result in the metric threshold alert details page, so I checked whether this value can be retrieved correctly there or not:
![image](https://user-images.githubusercontent.com/12370520/230380867-3a0520fd-687c-4d88-8161-278abfe8fc88.png)

**Note**
I will add tests later, I would like to get feedback about the implementation first.

## 🧪 How to test
- Add xpack.observability.unsafe.alertDetails.metrics.enabled: true to the Kibana config
- Create a metric threshold and inventory rule that generates an alert
- Check the alert document for the `kibana.alert.evaluation.values` field, it should be an array with the result of evaluation for the related criteria
- If you are using metricbeat, stop it so the value of evaluation will be null
- Go to the alert details page, you should be able to see the main chart even when the evaluation value is null
- Check the alert document for the `kibana.alert.evaluation.values` field, it should be an array including a null value
